### PR TITLE
Adding Conditional DCGAN_MNIST

### PR DESCRIPTION
**Pull request by AdarshKumar712**
This is an Implementation of Conditional DCGAN on MNIST Dataset. This I have implemented taking code-references from [DC-GAN](https://github.com/FluxML/model-zoo/tree/master/vision/dcgan_mnist) implementation available on model-zoo.

### CODE REVIEW

File: vision/cdcgan_mnist/cGAN_mnist.jl [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    epochs::Int = 25
+    verbose_freq::Int = 1000
+    output_x::Int = 6        # No. of sample images to concatenate along x-axis 
+    output_y::Int = 6        # No. of sample images to concatenate along y-axis
+    lr_dscr::Float64 = 0.0001
+    lr_gen::Float64 = 0.0001
+end
+
+
+mutable struct discriminator
```
Comment (CarloLucibello) [inline vision/cdcgan_mnist/cGAN_mnist.jl:32]: struct name should start with upper case

File: vision/cdcgan_mnist/cGAN_mnist.jl [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    m1 = Chain(Dense(args.nclasses,784), x -> reshape(x, 28,28,1,size(x,2))) |> gpu
+    model = Chain(Conv((3,3), 2=>128, pad=(1,1), stride = (2,2)),
+                  x->leakyrelu.(x, 0.2f0),
+                  Dropout(0.4),
+                  Conv((3,3), 128=>128, pad= (1,1), stride = (2,2), leakyrelu),
+                  x->leakyrelu.(x, 0.2f0),
+                  x -> reshape(x, :, size(x, 4)),
+                  Dropout(0.4),
+                  Dense(6272, 1)) |> gpu
+    discriminator(m1,model)
```
Comment (CarloLucibello) [inline vision/cdcgan_mnist/cGAN_mnist.jl:47]: please use whitespaces consistently

File: vision/cdcgan_mnist/cGAN_mnist.jl [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    output_x::Int = 6        # No. of sample images to concatenate along x-axis 
+    output_y::Int = 6        # No. of sample images to concatenate along y-axis
+    lr_dscr::Float64 = 0.0001
+    lr_gen::Float64 = 0.0001
+end
+
+
+mutable struct discriminator
+    m1
+    model
```
Comment (CarloLucibello) [inline vision/cdcgan_mnist/cGAN_mnist.jl:34]: these names are not very informative

**Comment by AdarshKumar712** [comment]
@CarloLucibello I have updated the changes as you said. Also, just as you mentioned in one other PR on `vae.jl` to remove `permutedims` when loading data with `MLDatasets` as data is already `WHCN` format, so I did the same with this one as well. 
One thing I wanted to ask if its right to keep the results of running the model(ie. images here) as well with the model code?

**Pull request merged**